packages/react-scripts/scripts/start.js

```diff
@@ -105,7 +122,7 @@ checkBrowsers(paths.appPath, isInteractive)
     // Create a webpack compiler that is configured with custom messages.
     const compiler = createCompiler({
       appName,
-      config,
+      config: [webConfig].concat(paths.appNodeIndexJs ? [nodeConfig] : []),
```

Comment:
```js
// maybe easier to avoid unnecessary concat
config: paths.appNodeIndexJs ? [webConfig, nodeConfig] : [webConfig],
```

I went down this route in my own ssr-enabled fork and ended up not using the multicompiler in normal day-to-day development. I was having issues where my large app was taking a _very_ long time to recompile. Doing `start` in client-only mode was super fast but multicompiler was slow (more than a minute to rebuild). I ended up adding a custom `watch.js` script that allowed me to watch the client and server code in separate processes, which was super fast.

I would suggest testing this on a large app to see if you're having the same issue.